### PR TITLE
Added coin toss and dice roll in cogs/misc.py

### PR DESCRIPTION
**Pull request by aryamaanthakur**
I've added a coin toss and dice roll feature for issue #147 

`t.coin` will show either a Head or a Tail and `t.coin n` will show the result of `n` tosses.
Similarly, `t.dice` will output a number between 1 & 6 and `t.dice n` will show the result of `n` dice rolls.

Please review my PR. Thank you!

### CODE REVIEW

**Comment by albertopoljak** [comment]
Looks good, there are just some small tiny things that could be done better.
I'll leave a few comments and you can change them if you wish.

File: bot/cogs/misc.py [hunk 1]
```diff
@@ -11,6 +11,8 @@
 from bot.utils.checks import check_if_it_is_tortoise_guild
 from bot.constants import embed_space, tortoise_paste_service_link
 
+from random import randint
```
Comment (albertopoljak) [inline bot/cogs/misc.py:14]: We categorize imports based on type, so it looks something like:
```
import ...  # built in modules

import ... # custom modules

import ... # project module
```
We also sort them by length, this makes it more clean and easier to follow. Especially for medium/big projects where you can have dozen of imports

File: bot/cogs/misc.py [hunk 2]
```diff
@@ -269,6 +271,27 @@ async def zen(self, ctx):
     async def antigravity(self, ctx):
         await ctx.send(embed=info("https://xkcd.com/353/", ctx.me, title=""))
 
+    @commands.command(aliases=['toss'])
```
Comment (albertopoljak) [inline bot/cogs/misc.py:274]: Use `"`for consistency.

File: bot/cogs/misc.py [hunk 3]
```diff
@@ -269,6 +271,27 @@ async def zen(self, ctx):
...
+        """Tosses a coin"""
+        sample_space = ["Head", "Tail"]
+        if times == 1:
+            coin_toss = sample_space[randint(0, 1)]
+            await ctx.send(embed=info(f":coin: | Coin Toss | **{coin_toss}**", ctx.me, title=""))
+        else:
+            coin_toss = ", ".join([sample_space[randint(0, 1)] for i in range(times)])
+            await ctx.send(embed=info(f":coin: | Coin tossed {times} times | **{coin_toss}**", ctx.me, title=""))
+
+    @commands.command(aliases=['roll'])
```
Comment (albertopoljak) [inline bot/cogs/misc.py:285]: Here also use `"` for consistency

File: bot/cogs/misc.py [hunk 4]
```diff
@@ -269,6 +271,27 @@ async def zen(self, ctx):
     async def antigravity(self, ctx):
         await ctx.send(embed=info("https://xkcd.com/353/", ctx.me, title=""))
 
+    @commands.command(aliases=['toss'])
+    async def coin(self, ctx, times: int = 1):
+        """Tosses a coin"""
+        sample_space = ["Head", "Tail"]
```
Comment (albertopoljak) [inline bot/cogs/misc.py:277]: For constants it's better to use tuples as, aside for being more logical/recommended, they are also more optimized and won't get constructed each time the method is called.

**Comment by aryamaanthakur** [comment]
Thanks a lot for the comments. I changed the code as instructed.

File: bot/cogs/misc.py [hunk 5]
```diff
@@ -269,6 +271,27 @@ async def zen(self, ctx):
...
+        if times == 1:
+            coin_toss = sample_space[randint(0, 1)]
+            await ctx.send(embed=info(f":coin: | Coin Toss | **{coin_toss}**", ctx.me, title=""))
+        else:
+            coin_toss = ", ".join([sample_space[randint(0, 1)] for i in range(times)])
```
Comment (albertopoljak) [inline bot/cogs/misc.py:281]: You can use a single underscore  `_` for dummy/throwaway variables. Same for the other command. It's not really important but its the convention we use. 

There is also no need to construct a list here as join can take a generator as well.

File: bot/cogs/misc.py [hunk 6]
```diff
@@ -269,6 +271,27 @@ async def zen(self, ctx):
     async def antigravity(self, ctx):
         await ctx.send(embed=info("https://xkcd.com/353/", ctx.me, title=""))
 
+    @commands.command(aliases=['toss'])
+    async def coin(self, ctx, times: int = 1):
```
Comment (albertopoljak) [inline bot/cogs/misc.py:274]: Maybe add a check to prevent users to pass large `times`, as it would be spam and could error if message length is exceeded.

**Comment by albertopoljak** [comment]
Ah you were to fast I didn't finish yet, added few more.
It's  just minor things the PR is good as it is, so do them if you feel like it, I'll merge for sure later.

File: bot/cogs/misc.py (hunk 6 above)
Comment (aryamaanthakur) [reply bot/cogs/misc.py:274]: Shall I set the max limit of `times` to 25?

Comment (albertopoljak) [reply bot/cogs/misc.py:274]: > 
> 
> Shall I set the max limit of `times` to 25?

Sounds good

**Comment by aryamaanthakur** [comment]
Sorry for multiple commits. I got too excited I guess XD

**Comment by albertopoljak** [comment]
Looks good, thank you very much!

I'll merge it now. This is my first merge as a Hacktoberfest maintainer and I think I've set it up correctly, but if it for some reason doesn't show up in your Hacktoberfest profile be sure to yell at me here later! :)

**Pull request merged**